test/test_regression.py

```diff
@@ -0,0 +1,312 @@
+""" Regression tests for https://github.com/nameko/nameko/issues/428
```

Comment:
if the problem was that blocking in `handle_message` blocks the whole queue consumer, maybe it would be enough to check that that is no longer happening, rather than explicitly checking that heartbeats still work?